com.unity.perception/Runtime/GroundTruth/Labelers/KeypointLabeler.cs

```diff
@@ -131,8 +133,11 @@ int DetermineKeypointState(Keypoint keypoint, Color32 instanceIdColor, (int x, i
         {
             if (keypoint.state == 0) return 0;
 
-            var centerX = Mathf.RoundToInt(keypoint.x);
-            var centerY = Mathf.RoundToInt(keypoint.y);
+            var centerX = Mathf.FloorToInt(keypoint.x);
+            var centerY = Mathf.FloorToInt(keypoint.y);
+
+            if (!PixelOnScreen(centerX, centerY, dimensions))
+                return 0;
```

Comment:
So we return early if the center point is not inside the image dimensions. Later we check around the center with a tolerance. Doesn't this check here render that later check pointless?

Reply:
Right on all counts. The tolerance checks below are there to ensure that we are looking in the right places in the image for object visibility when the keypoint is in the frame. This is a slight change in design from before, since now we are saying that only keypoints that land in the image should ever be reported.

Reply:
Aaa I see. Makes sense now.